pkg/instance/instance.go

```diff
@@ -447,18 +455,40 @@ func (s *Server) watchSPDKReplica(ctx context.Context, req *emptypb.Empty, srv r
 		return errors.Wrap(err, "failed to create SPDK replica watch notifier")
 	}
 
+	failureCount := 0
 	for {
+		if failureCount >= maxMonitorRetryCount {
+			logrus.Errorf("Continuously receiving errors for %v times, stopping watching SPDK replicas", maxMonitorRetryCount)
+			return fmt.Errorf("continuously receiving errors for %v times, stopping watching SPDK replicas", maxMonitorRetryCount)
+		}
+
 		select {
 		case <-ctx.Done():
-			logrus.Info("Stopped watching SPDK replica")
-			return nil
+			logrus.Info("Stopped watching SPDK replicas")
+			return ctx.Err()
 		default:
-			_, err := notifier.Recv()
-			if err != nil {
-				logrus.WithError(err).Error("Failed to receive next item in SPDK replica watch")
-			} else {
-				if err := srv.Send(&empty.Empty{}); err != nil {
-					return errors.Wrap(err, "failed to send instance response")
+			// Use a channel to receive the result of notifier.Recv()
+			notifyRecvChan := make(chan error, 1)
+
+			go func() {
+				_, err := notifier.Recv()
+				notifyRecvChan <- err
```

Comment:
NIT: close the channel after this sending. 
Same below.

Reply:
Addressed. Thank you.